core/shared/platform/esp-idf/espidf_memmap.c

```diff
@@ -9,7 +9,12 @@
 void *
 os_mmap(void *hint, size_t size, int prot, int flags)
 {
-    return os_malloc((int)size);
+    if (prot & MMAP_PROT_EXEC) {
+        return heap_caps_malloc(size, MALLOC_CAP_EXEC);
```

Comment:
Not sure whether the memory allocated by heap_caps_malloc can be freed by os_free?  Another issue is that will the return add be 8-byte aligned? Note that in os_malloc, we allocate extra bytes to make it 8-byte aligned.

Reply:
> Not sure whether the memory allocated by heap_caps_malloc can be freed by os_free?

This is in fact possible, quoting the [official documentation:](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/mem_alloc.html#memory-capabilities)

> Because `malloc` uses the capabilities-based allocation system, memory allocated using `heap_caps_malloc()` can be freed by calling the standard `free()` function.

Reply:
Yes, @1c3t3a, but it was still an error due to the extra metadata. Working on a fix with the actual layout..